presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -333,10 +306,6 @@ else if (pendingSplits.isEmpty()) {
             }
         }
 
-        if (autoDropCompletedLifespans) {
-            drainCompletedLifespans();
-        }
```

Comment:
Moving this to the end potentially changes logic because it will affect the evaluation of the if condition immediately down below.